bvm/ballerina-logging/src/main/java/org/ballerinalang/logging/BLogManager.java

```diff
@@ -127,6 +131,31 @@ public void setHttpTraceLogHandler() throws IOException {
         httpTraceLogger.setLevel(Level.FINEST);
     }
 
+    /**
+     * Sets the Http access log
+     * @param arg the argument for access logging
+     * @throws IOException if there are IO problems opening the files.
+     */
+    public void setHttpAccessLogHandler(String arg) throws IOException {
+        Handler handler;
+        if (arg.equalsIgnoreCase("console")) {
+            handler = new ConsoleHandler();
+        } else {
+            handler = new FileHandler(arg, true);
```

Comment:
What happens if the user gives an invalid path? I think we may have to handle it

Reply:
Handled at the point where the method is called in `BLogManager`

Reply:
I think you need to think this through. I tried running with the path as `logs/access.log` where `logs` is a non-existent directory and it just failed without a proper error message. This only happens when the error is not handled properly. Given below is the log.
```
[2018-03-06 17:37:36,962] ERROR {org.ballerinalang.launcher.Main} - ballerina: internal error occurred 
java.lang.ExceptionInInitializerError
	at org.ballerinalang.net.http.HttpUtil.getDefaultOrDynamicListenerConfig(HttpUtil.java:625)
	at org.ballerinalang.net.http.HTTPServicesRegistry.registerService(HTTPServicesRegistry.java:97)
	at org.ballerinalang.net.http.BallerinaHttpServerConnector.serviceRegistered(BallerinaHttpServerConnector.java:58)
	at org.ballerinalang.connector.impl.ServerConnectorRegistry.registerService(ServerConnectorRegistry.java:88)
	at org.ballerinalang.BLangProgramRunner.deployService(BLangProgramRunner.java:107)
	at org.ballerinalang.BLangProgramRunner.runService(BLangProgramRunner.java:73)
	at org.ballerinalang.launcher.LauncherUtils.runServices(LauncherUtils.java:120)
	at org.ballerinalang.launcher.LauncherUtils.runProgram(LauncherUtils.java:95)
	at org.ballerinalang.launcher.Main$RunCmd.execute(Main.java:258)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.launcher.Main.main(Main.java:63)
Caused by: org.ballerinalang.connector.api.BallerinaConnectorException: Error in configuring HTTP access log
	at org.ballerinalang.net.http.HttpConnectionManager.<init>(HttpConnectionManager.java:86)
	at org.ballerinalang.net.http.HttpConnectionManager.<clinit>(HttpConnectionManager.java:58)
	... 11 more
Caused by: java.nio.file.NoSuchFileException: logs/access.log.lck
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:177)
	at java.nio.channels.FileChannel.open(FileChannel.java:287)
	at java.nio.channels.FileChannel.open(FileChannel.java:335)
	at java.util.logging.FileHandler.openFiles(FileHandler.java:459)
	at java.util.logging.FileHandler.<init>(FileHandler.java:326)
	at org.ballerinalang.logging.BLogManager.setHttpAccessLogHandler(BLogManager.java:144)
	at org.ballerinalang.net.http.HttpConnectionManager.<init>(HttpConnectionManager.java:84)
	... 12 more
```

Reply:
I am rethrowing a `BallerinaConnectorException: Error in configuring HTTP access log`
Do you mean that I should print to the console as in [1] before rethrowing the error?
[1] https://github.com/ballerina-lang/ballerina/blob/aaf125661b92ebd05607abeb2ca304a73b47c8e4/cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java#L120

Reply:
No I meant, this exception doesn't seem to be handled correctly. When running it, try giving an invalid path. It results in `ballerina: internal error occurred`. This means an exception propagated all the way up to the root error handler, which shouldn't happen in this case.

Reply:
Fixed